modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/H2TableDescriptor.java

```diff
@@ -464,6 +456,29 @@ private Index createHashIndex(GridH2Table tbl, List<IndexColumn> cols) {
         return null;
     }
 
+    /**
+     * Create Lucene index.
+     *
+     * @param cacheName Cache name.
+     * @param log Logger.
+     * @return Index.
+     */
+    private LuceneIndex createLuceneIndex(@Nullable String cacheName, IgniteLogger log) {
+        LuceneIndexFactory[] ext = idx.kernalContext().plugins().extensions(LuceneIndexFactory.class);
+
+        if (nonNull(ext)) {
+            if (ext.length > 1)
+                log.info("More than one LuceneIndexFactory extension is defined.");
+
+            LuceneIndexFactory factory = ext[0];
+
+            return factory.create(idx.kernalContext(), cacheName, type);
+        }
+
+        throw new IgniteException("Failed to create index because Lucene module is disabled (consider adding module " +
+            "gridgain-lucene to classpath or moving it from 'optional' to 'libs' folder).");
```

Comment:
To make this look better, it should be explicitly highlighted that the feature is only available in EE/UE, and then recommending to upgrade.

Consider referring to the feature docs that should be available later.

Something like
"Failed to create _[which?]_ index. The feature _[name]_ requires the `gridgain-lucene` module available for GridGain Enterprise and GridGain Ultimate editions. Learn more: _[link to vector search docs, I guess?]_"

The wording is not exact, play with it. I just wanted to provide an idea.
Also, you'd like to sync that with @isapego who's solving the same task because he limits a private vector-search feature in a public Python client.

Reply:
The above is about the case when GG CE is used.
Another case is when a user is already on GG EE / UE, proceed with the usual technical message to clarify that the module should be included, e.g. what we currently have in this PR.